src/main/web/src/scenes/Profile/index.tsx

```diff
@@ -1,52 +1,133 @@
-import React, {useRef, useState} from "react";
+import React, { useEffect, useRef, useState } from "react";
 import DOMPurify from "dompurify";
 import "./index.css";
-import {Header} from "../../organisms/header/Header";
-import {Footer} from "../../organisms/footer/Footer";
-import {logout} from "../../services/AuthService";
-import {NavigateFunction, useNavigate} from "react-router-dom";
+import { Header } from "../../organisms/header/Header";
+import { Footer } from "../../organisms/footer/Footer";
+import { getUserId, logout } from "../../services/AuthService";
+import { NavigateFunction, useNavigate } from "react-router-dom";
 import ScrollUpButton from "../../atoms/ScrollUpButton";
 import AdBlockOverlay from "../../organisms/ad-block-overlay/AdBlockOverlay";
-import {useDetectAdBlock} from "adblock-detect-react";
-import {usePreventScrolling} from "../../organisms/ad-block-overlay/preventScrolling";
+import { useDetectAdBlock } from "adblock-detect-react";
+import { usePreventScrolling } from "../../organisms/ad-block-overlay/preventScrolling";
+import {fetchUserData, updatePassword} from "../../services/ProfileDataService";
+import { fetchUserImage } from "../../services/ProfilePictureService";
+import { updateAge, updateDescription, updateCourse, updateEmail, updateUsername, updatePicture } from "../../services/ProfileDataService";
 
-interface ProfileData {
+interface UserData {
```

Comment:
Best Practise: Extract this to a seperate file called `UserData` in a folder named `models`